0-bootstrap/modules/jenkins/main.tf

```diff
@@ -112,7 +125,147 @@ resource "google_compute_firewall" "allow_ssh_to_jenkins_agent_fw" {
 
 resource "google_compute_network" "jenkins_agents" {
   project = module.cicd_project.project_id
-  name    = "jenkins-agents-network"
+  name    = "vpc-b-jenkinsagents"
 }
 
 // TODO(caleonardo): add an option to create a Jenkins Master in GCP, in case the user doesn't have one on-prem
+// If you do not have a Jenkins implementation and don't want one, you should use the Cloud Build module instead of this Jenkins module
+
+/******************************************
+  VPN Connectivity on-prem <--> CICD project
+*******************************************/
+
+// TODO(Amanda / Daniel): add VPN for connectivity between Jenkins Master on prem with Jenkins Agent in GCP
+
+/******************************************
+  Jenkins IAM for admins
+*******************************************/
+
+resource "google_project_iam_member" "org_admins_jenkins_admin" {
+  project = module.cicd_project.project_id
+  role    = "roles/compute.admin"
+  member  = "group:${var.group_org_admins}"
+}
+
+resource "google_project_iam_member" "org_admins_jenkins_viewer" {
+  project = module.cicd_project.project_id
+  role    = "roles/viewer"
+  member  = "group:${var.group_org_admins}"
+}
+
+/******************************************
+  Jenkins Artifact bucket
+*******************************************/
+
+resource "google_storage_bucket" "gcs_jenkins_artifacts" {
+  project            = module.cicd_project.project_id
+  name               = format("%s-%s-%s-%s", var.storage_bucket_prefix, module.cicd_project.project_id, "jenkins-artifacts", random_id.suffix.hex)
+  location           = var.default_region
+  labels             = var.storage_bucket_labels
+  bucket_policy_only = true
+  versioning {
+    enabled = true
+  }
+}
+
+// TODO(Rohan): Evaluate if / how to migrate this to Secrets Manager
+/******************************************
+  KMS Keyring configuration -
+  to protect any secrets in CICD project
+*****************************************/
+
+resource "google_kms_key_ring" "tf_keyring" {
+  project  = module.cicd_project.project_id
+  name     = "tf-keyring"
+  location = var.default_region
+  depends_on = [
+    google_project_service.jenkins_apis,
+  ]
+}
+
+/******************************************
+  KMS Key
+ *****************************************/
+
+resource "google_kms_crypto_key" "tf_key" {
+  name     = "tf-key"
+  key_ring = google_kms_key_ring.tf_keyring.self_link
+}
+
+/******************************************
+  Permissions to decrypt.
+ *****************************************/
+
+resource "google_kms_crypto_key_iam_binding" "jenkins_crypto_key_decrypter" {
+  crypto_key_id = google_kms_crypto_key.tf_key.self_link
+  role          = "roles/cloudkms.cryptoKeyDecrypter"
+
+  members = [
+    "serviceAccount:${google_service_account.jenkins_agent_gce_sa.email}",
+    "serviceAccount:${var.terraform_sa_email}"
+  ]
+  depends_on = [
```

Comment:
explicit dependency can be removed if using `activate_apis` here and throughout

Reply:
Yes, this is removed now. Thank you.